### ISSUE TITLE
crash with vim.regex call in user command preview with invalid regex

### ISSUE THREAD

**Issue by grassdne** [neovim/neovim#20546]
### Neovim version (nvim -v)

0.8.0

### Vim (not Nvim) behaves the same?

/

### Operating system/version

openSUSE Tumbleweed on Windows 11

### Terminal name/version

Windows Terminal

### $TERM environment variable

xterm-256color

### Installation

Zypper

### How to reproduce the issue

```lua
nvim --clean
:lua vim.api.nvim_create_user_command("Crash", function() end, {preview=function() pcall(vim.regex, "\\z") end})
:Crash
```

### Expected behavior

nothing, in this example.

### Actual behavior

nvim crashes as you type out :Crash
> nvim: /home/abuild/rpmbuild/BUILD/neovim-0.8.0/src/nvim/lua/stdlib.c:308: nlua_regex: Assertion `prog' failed.
>                                                                                                                [1]    11857 IOT instruction  nvim --clean

**Comment by grassdne** [comment]
This is significant because because previewing the result of a regular expression while the user types out \zs will crash Neovim.

**Comment by zeertzjq** [comment]
Hmm, I cannot reproduce.

**Comment by famiu** [comment]
I cannot reproduce the issue either.

**Comment by clason** [comment]
@grassdne can you share the full stacktrace?

**Comment by grassdne** [comment]
Oh... goddamn markdown escaped my backslashes!
I put the commands in a code block.

**Comment by zeertzjq** [comment]
Oh, so the same crash happens with `:silent! lua vim.regex('\\z')`

**Comment by grassdne** [comment]
quick fix #20555
